packages/spidermonkey.rb

```diff
@@ -0,0 +1,62 @@
+require 'package'
+
+class Spidermonkey < Package
+  description 'SpiderMonkey is Mozilla\'s JavaScript engine written in C and C++.'
+  homepage 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey'
+  version '84.0'
+  compatibility 'all'
+  source_url 'https://archive.mozilla.org/pub/firefox/releases/84.0/source/firefox-84.0.source.tar.xz'
+  source_sha256 '23273ef0165b243f5d0908c38e7854d38070282c9b526e8d93b7503cd5f69138'
+
+  depends_on 'yasm'
```

Comment:
I'm guessing this is a build dependency also?

Reply:
considering that yasm only provides binaries, header files and one static library, you're probably right. I'll test that to make sure and get back in an hour when it builds.

Reply:
looks like yasm is a build dep, i'll make it so